modules/ui/fields/input.js

```diff
@@ -167,9 +168,29 @@ export function uiFieldText(field, context) {
                     if (value) window.open(value, '_blank');
                 })
                 .merge(outlinkButton);
+        } else if (field.key.includes('colour')) {
+            input.attr('type', 'text');
+
+            updateColourPreview();
         }
     }
 
+    function updateColourPreview() {
+        wrap.selectAll('.foreign-id-permalink')
+            .remove();
+
+        outlinkButton = wrap.selectAll('.foreign-id-permalink')
+            .data([utilGetSetValue(input)], d => d);
+
+        outlinkButton
+            .enter()
+            .append('div')
+            .attr('class', 'form-field-button foreign-id-permalink colour-preview')
+            .append('div')
+            .style('background-color', d => d)
```

Comment:
This isn’t far from a [color input field](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/color), which would make it easier for the user to choose an sRGB value. However, some browsers render this field as a color well and don’t fall back gracefully when the value is a color keyword.

Reply:
I experimented with [a colour picker](https://github.com/openstreetmap/iD/compare/3a3d97...54d922), but there are some issues:
- as you said, the colour picker doesn't work when value is a color keyword
- the default `<input type="color" />` element shows black for invalid colours or an empty input field, which would be quite confusing 
- the default `<input type="color" />` element doesn't have a submit button, so iD's text field would only update when the user clicks out of the colour picker. This would be confusing for some people

So a think the simple colour preview is the better option